hirosystems/stacks-blockchain-api/templates/api-writer/statefulset.yaml

```diff
@@ -209,50 +211,61 @@ spec:
-            - name: PG_DATABASE
-              value: {{ include "postgresql.database" .Subcharts.postgresql | quote }}
-            - name: PG_SCHEMA
-              value: {{ include "postgresql.database" .Subcharts.postgresql }}
-            - name: PG_CONNECTION_POOL_MAX
-              value: {{ .Values.apiWriter.config.pgConnectionPoolMax | quote }}
             - name: PG_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  key: {{ include "postgresql.adminPasswordKey" .Subcharts.postgresql | quote }}
-                  name: {{ include "postgresql.secretName" .Subcharts.postgresql | quote }}
+                  key: {{ include "postgresql.adminPasswordKey" .Subcharts.postgresql }}
+                  name: {{ include "postgresql.secretName" .Subcharts.postgresql }}
+            - name: PG_APPLICATION_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: PG_CONNECTION_POOL_MAX
+              value: {{ default "50" .Values.apiWriter.config.pgConnectionPoolMax | quote }}
+            {{- if .Values.cdn.enabled }}
             - name: PG_IDENT_INDEX_TYPE
               value: hash
-            - name: NODE_ENV
-              value: production
-            - name: BNS_IMPORT_DIR
-              value: {{ .Values.apiWriter.persistence.bns.mountPath }}
-            - name: STACKS_API_EVENT_REPLAY_LOG_LEVEL
-              value: debug
-            - name: STACKS_API_ENABLE_FT_METADATA
-              value: {{ ternary "1" "0" .Values.apiWriter.config.enableFtMetadata | quote }}
-            - name: STACKS_API_ENABLE_NFT_METADATA
-              value: {{ ternary "1" "0" .Values.apiWriter.config.enableNftMetadata | quote }}
+            {{- if .Values.apiWriter.extraEnvVars }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.apiWriter.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- if .Values.apiWriter.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.apiWriter.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.apiWriter.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.apiWriter.extraEnvVarsSecret "context" $) }}
+            {{- end }}
           command:
             - sh
             - -c
-            {{- if .Values.apiWriter.config.replayEvents.prunedMode }}
-            - |
-              node ./lib/index.js import-events --file ${STACKS_EXPORT_EVENTS_FILE} --wipe-db --force --mode pruned
-            {{- else }}
-            - |
-              node ./lib/index.js import-events --file ${STACKS_EXPORT_EVENTS_FILE} --wipe-db --force
-            {{- end }}
+              {{- printf "node ./lib/index.js from-parquet-events --workers=%d" .Values.apiWriter.replayEvents.workers | quote }}
```

Comment:
fixed this.
the dry-run output is as follows:
```
        - name: replay-events
          image: docker.io/hirosystems/stacks-blockchain-api:7.3.2
          imagePullPolicy:
          env:
            - name: STACKS_API_LOG_LEVEL
              value: "debug"
            - name: STACKS_CHAIN_ID
              value: "0x00000001"
            - name: STACKS_EVENTS_DIR
              value: /hirosystems/api-writer/data/events
            - name: PG_DATABASE
              value: stacks_blockchain_api
            - name: PG_SCHEMA
              value: stacks_blockchain_api
            - name: PG_PRIMARY_HOST
              value: stacks-blockchain-api-5-1700233640-postgresql
            - name: PG_HOST
              value: stacks-blockchain-api-5-1700233640-postgresql-all
            - name: PG_PORT
              value: "5432"
            - name: PG_USER
              value: postgres
            - name: PG_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: "postgres-password"
                  name: "stacks-blockchain-api-5-1700233640-postgresql"
            - name: PG_APPLICATION_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: PG_CONNECTION_POOL_MAX
              value: "50"
          envFrom:
          command:
              - node ./lib/index.js from-parquet-events --workers=4
          securityContext:
            readOnlyRootFilesystem: false
            runAsNonRoot: true
            runAsUser: 1001
          resources:
            limits: {}
            requests: {}
          volumeMounts:
            - name: data
              mountPath: /hirosystems/api-writer/data
            - name: bns-data
              mountPath: /hirosystems/api-writer/bns
```

Reply:
Nice, much better! Have you tested if that command syntax works? I'm not sure the command will run because I believe k8s will send the entire `node ./lib/index.js from-parquet-events --workers=4` to the shell as a single string. Typically commands with arguments need to be broken over multiple entries in the `command` list. Ideally, this would be split into `command` and `args` like so:
```
          command:
              - node
          args:
              - ./lib/index.js
              - from-parquet-events
              - --workers=4
```

Reply:
fixed. and added additional arg for `--max-old-space-size`
the output now looks like this
```
        - name: replay-events
          image: docker.io/hirosystems/stacks-blockchain-api:7.3.2
          imagePullPolicy:
          env:
            - name: --max-old-space-size
              value: 8192
            - name: STACKS_API_LOG_LEVEL
              value: "info"
            - name: STACKS_CHAIN_ID
              value: "0x00000001"
            - name: STACKS_EVENTS_DIR
              value: /hirosystems/api-writer/data/events
            - name: PG_DATABASE
              value: stacks_blockchain_api
            - name: PG_SCHEMA
              value: stacks_blockchain_api
            - name: PG_PRIMARY_HOST
              value: stacks-blockchain-api-5-1700244264-postgresql
            - name: PG_HOST
              value: stacks-blockchain-api-5-1700244264-postgresql-all
            - name: PG_PORT
              value: "5432"
            - name: PG_USER
              value: postgres
            - name: PG_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: "postgres-password"
                  name: "stacks-blockchain-api-5-1700244264-postgresql"
            - name: PG_APPLICATION_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: PG_CONNECTION_POOL_MAX
              value: "50"
          envFrom:
          command:
          - node
          args:
          - "./lib/index.js"
          - "from-parquet-events"
          - "--workers=4"
          - "--max-old-space-size=8192"
          securityContext:
            readOnlyRootFilesystem: false
            runAsNonRoot: true
            runAsUser: 1001
          resources:
            limits: {}
            requests:
              memory: 32Gi
          volumeMounts:
            - name: data
              mountPath: /hirosystems/api-writer/data
            - name: bns-data
              mountPath: /hirosystems/api-writer/bns
```